xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -14776,7 +14776,7 @@ glusterd_check_brick_order(dict_t *dict, char *err_str, int32_t type,
         if (tmpptr == NULL)
             goto check_failed;
         addrlen = strlen(brick) - strlen(tmpptr);
-        strncpy(brick_addr, brick, addrlen);
+        memcpy(brick_addr, brick, addrlen);
```

Comment:
I think the warning is that there is no bounds checking to verify that`addrlen` is less than size of `brick_addr` which is a 128 byte array. Commit 6a14d2b3e33aa59e9b14bdbb4395fb529c3c2616 recently fixed a similar issue.